PrefabDetachPrefabAndRemoveContainerTests encounters a C2220 error during compilation

**Describe the bug**
The PrefabDetachPrefabAndRemoveContainerTests file has several unreferenced parameters which is being flagged as a error during compilation with the release target

**Assets required**
Link to Jenkins run here: https://jenkins.build.o3de.org/blue/rest/organizations/jenkins/pipelines/O3DE/branches/development/runs/4971/nodes/555/steps/571/log/?start=0

**Steps to reproduce**
1. Build O3DE with the release target: `cmake --build . --target ALL_BUILD --config release  -- /m /nologo`

**Expected behavior**
Compilation completes without error

**Actual behavior**
The PrefabDetachPrefabAndRemoveContainerTests.cpp file encounter a C2220 warning as error and C4100 warnings 
Note: This should have failed the Jenkins build but it is currently passing. Currently investigating this here: 

**Screenshots/Video**
N/A

**Found in Branch**
Development

**Commit ID from [o3de/o3de](https://github.com/o3de/o3de) Repository**
75bbcc1ae57670ca5e2042e8a6d10cd26fc9e07d

**Desktop/Device (please complete the following information):**
 - Device: AWS EC2
 - OS: Windows Server 
 - Version: 2022 (eqv to Windows 11)
 - CPU: AMD EPYC 7R13
 - GPU: N/A
 - Memory 32GB

**Additional context**
Log message as follows:

```
[2024-04-20T16:18:48.107Z] D:/workspace/o3de/Code/Framework/AzToolsFramework/Tests/Prefab/PrefabDetachPrefabAndRemoveContainerTests.cpp(447,65): error C2220: the following warning is treated as an error [D:\workspace\o3de\build\windows\Code\Framework\AzToolsFramework\AzToolsFramework.Tests.vcxproj]
[2024-04-20T16:18:48.107Z]           levelInstance->get().GetEntityIds([&count](AZ::EntityId entityId) { count++; return true; });
[2024-04-20T16:18:48.365Z]                                                                   ^
[2024-04-20T16:18:49.296Z] D:/workspace/o3de/Code/Framework/AzToolsFramework/Tests/Prefab/PrefabDetachPrefabAndRemoveContainerTests.cpp(447,65): warning C4100: 'entityId': unreferenced formal parameter [D:\workspace\o3de\build\windows\Code\Framework\AzToolsFramework\AzToolsFramework.Tests.vcxproj]
[2024-04-20T16:18:49.554Z] D:/workspace/o3de/Code/Framework/AzToolsFramework/Tests/Prefab/PrefabDetachPrefabAndRemoveContainerTests.cpp(451,87): warning C4100: 'instance': unreferenced formal parameter [D:\workspace\o3de\build\windows\Code\Framework\AzToolsFramework\AzToolsFramework.Tests.vcxproj]
[2024-04-20T16:18:49.812Z]           levelInstance->get().GetNestedInstances([&count](AZStd::unique_ptr<Instance>& instance) { count++; });
[2024-04-20T16:18:50.069Z]                                                                                         ^
[2024-04-20T16:18:50.326Z] D:/workspace/o3de/Code/Framework/AzToolsFramework/Tests/Prefab/PrefabDetachPrefabAndRemoveContainerTests.cpp(462,72): warning C4100: 'entityId': unreferenced formal parameter [D:\workspace\o3de\build\windows\Code\Framework\AzToolsFramework\AzToolsFramework.Tests.vcxproj]
[2024-04-20T16:18:50.583Z]   Copying AWSCore.ResourceMappingTool runtime dependencies to output...
[2024-04-20T16:18:50.583Z]           levelInstance->get().GetEntityIds([&count](const AZ::EntityId& entityId) { count++; return true; });
[2024-04-20T16:18:50.840Z]      Creating library D:/workspace/o3de/build/windows/lib/release/AWSCore.Editor.Tests.lib and object D:/workspace/o3de/build/windows/lib/release/AWSCore.Editor.Tests.exp
[2024-04-20T16:18:50.840Z]                                                                          ^
[2024-04-20T16:18:51.404Z] D:/workspace/o3de/Code/Framework/AzToolsFramework/Tests/Prefab/PrefabDetachPrefabAndRemoveContainerTests.cpp(466,87): warning C4100: 'instance': unreferenced formal parameter [D:\workspace\o3de\build\windows\Code\Framework\AzToolsFramework\AzToolsFramework.Tests.vcxproj]
[2024-04-20T16:18:51.662Z]           levelInstance->get().GetNestedInstances([&count](AZStd::unique_ptr<Instance>& instance) { count++; });
[2024-04-20T16:18:52.226Z]                                                                                         ^
```

--- a/AutomatedTesting/Gem/PythonTests/editor/TestSuite_Main.py
+++ b/AutomatedTesting/Gem/PythonTests/editor/TestSuite_Main.py
@@ -58,6 +58,7 @@ class TestAutomationNoAutoTestMode(EditorTestSuite):
 
         from .EditorScripts import BasicEditorWorkflows_LevelEntityComponentCRUD as test_module
 
+    @pytest.mark.skip(reason="Skipped for intermittently failing.")
     class test_InputBindings_Add_Remove_Input_Events(EditorBatchedTest):
         from .EditorScripts import InputBindings_Add_Remove_Input_Events as test_module
 

--- a/Code/Framework/AzToolsFramework/Tests/Prefab/PrefabDetachPrefabAndRemoveContainerTests.cpp
+++ b/Code/Framework/AzToolsFramework/Tests/Prefab/PrefabDetachPrefabAndRemoveContainerTests.cpp
@@ -444,11 +444,11 @@ namespace UnitTest
 
         // before we detach, the level should contain 1 entity (the container entity) and 1 instance (of the car)
         int count = 0;
-        levelInstance->get().GetEntityIds([&count](AZ::EntityId entityId) { count++; return true; });
+        levelInstance->get().GetEntityIds([&count](AZ::EntityId) { count++; return true; });
         EXPECT_EQ(count, 1); // expect 1 real entity, that is, the container for the car.
 
         count = 0;
-        levelInstance->get().GetNestedInstances([&count](AZStd::unique_ptr<Instance>& instance) { count++; });
+        levelInstance->get().GetNestedInstances([&count](AZStd::unique_ptr<Instance>&) { count++; });
         EXPECT_EQ(count, 1); // expect 1 instance of another prefab (The the car)
 
         // Detach the car prefab.
@@ -459,11 +459,11 @@ namespace UnitTest
 
         // after we detach, the level should contain 3 entities (2 of them real entities, one of them an instance container)
         count = 0;
-        levelInstance->get().GetEntityIds([&count](const AZ::EntityId& entityId) { count++; return true; });
+        levelInstance->get().GetEntityIds([&count](const AZ::EntityId&) { count++; return true; });
         EXPECT_EQ(count, 3); // expect 2 REAL entities.
 
         count = 0;
-        levelInstance->get().GetNestedInstances([&count](AZStd::unique_ptr<Instance>& instance) { count++; });
+        levelInstance->get().GetNestedInstances([&count](AZStd::unique_ptr<Instance>&) { count++; });
         EXPECT_EQ(count, 1); // expect 1 instance of another prefab (the wheel)
 
         // Validate child entity order under the level after detaching the car prefab.  Should be engine, wheel, battery.

